File: LeetCode-Programs/Determine if String Halves Are Alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        mid=len(s)//2
        count_l, count_r= 0, 0
        for i in range(0,mid):
            if s[i]=='a' or s[i]=='e' or s[i]=='i' or s[i]=='o' or s[i]=='u' or s[i]=='A' or s[i]=='E' or s[i]=='I' or s[i]=='O' or s[i]=='U':
                count_l+=1
        
        for i in range(mid,len(s)):
            if s[i]=='a' or s[i]=='e' or s[i]=='i' or s[i]=='o' or s[i]=='u' or s[i]=='A' or s[i]=='E' or s[i]=='I' or s[i]=='O' or s[i]=='U':
                count_r+=1
                
        return count_l == count_r

File: LeetCode-Programs/Determine if String Halves Are Alike.cpp
class Solution {
public:
    bool halvesAreAlike(string s) {
        int mid=s.size()/2;
        int count_l=0, count_r=0;
        
        for(int i=0;i<mid;i++){
            if(s[i]=='a' || s[i]=='e' || s[i]=='i' || s[i]=='o' || s[i]=='u' || s[i]=='A' || s[i]=='E' || s[i]=='I' || s[i]=='O' || s[i]=='U')
                count_l++;
        }
        
        for(int i=mid;i<s.size();i++){
            if(s[i]=='a' || s[i]=='e' || s[i]=='i' || s[i]=='o' || s[i]=='u' || s[i]=='A' || s[i]=='E' || s[i]=='I' || s[i]=='O' || s[i]=='U')
                count_r++;
        }
        return count_l==count_r;
    }
};
